src/EscapeAnalysis.jl

```diff
@@ -136,26 +136,16 @@ end
 # analysis
 # ========
 
-# TODO flow-sensitivity
-#    Main motivation here is finalizer elision, e.g.:
-#    ```julia
-#    let
-#        m = make_m()
-#        if cond
-#            return nothing # <= insert a finalizer call here
-#        end
-#        return m
-#    end
-#    ```
-#    We can do setup separate states for each statement (probably better with sparse states
-#    since there can be many statements), or encoding the quivalent information into the lattice
-
 """
     x::EscapeLattice
 
 A lattice for escape information, which holds the following properties:
 - `x.Analyzed::Bool`: not formally part of the lattice, indicates this statement has not been analyzed at all
-- `x.ReturnEscape::Bool`: indicates it will escape to the caller via return (possibly as a field)
+- `x.ReturnEscape::BitSet`: keeps SSA numbers of return statements where it can be returned to the caller
+  * `isempty(x.ReturnEscape)` means it never escapes to the caller
+  * otherwise it indicates it will escape to the caller via return (possibly as a field),
+    where `0 ∈ x.ReturnEscape` has the special meaning that it's visible to the caller
+    simply because it's passed as call argument
```

Comment:
This sounds good. I suspect at some point we will want to split this to have a separate bit for each argument (so `1:nargs` is for arguments, and `nargs + pc:end` for other locations), but not yet necessary